orttraining/orttraining/python/training/onnxblock/blocks.py

```diff
@@ -427,3 +428,45 @@ def build(self, cast_input_name: str):
         self.base.graph.node.append(cast_node)
 
         return cast_output_name
+
+
+class Gemm(Block):
```

Comment:
I think I can add those on as needed basis.

Since the use case for this `Gemm` is similar to `torch.nn.Linear` (maybe I should call it Linear instead of Gemm). Since the users are introducing a new node in the graph, they only specify the `in_features` and the `out_features`. As such, they don't get to chose the shape of the weight. So `transB` is not very meaningful.

As for `transA`, it is mostly needed for optimizing the graph. I can add this as needed later on as needed.